sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnInvokers.java

```diff
@@ -0,0 +1,460 @@
+      return instrumentedType;
+    }
+
+    @Override
+    public ByteCodeAppender appender(final Target implementationTarget) {
+      return new ByteCodeAppender() {
+        @Override
+        public Size apply(
+            MethodVisitor methodVisitor,
+            Context implementationContext,
+            MethodDescription instrumentedMethod) {
+          StackManipulation manipulation =
+              new StackManipulation.Compound(
+                  // Push "this" reference to the stack
+                  MethodVariableAccess.REFERENCE.loadOffset(0),
+                  // Access the delegate field of the the invoker
+                  FieldAccess.forField(delegateField).getter(),
+                  invokeTargetMethod(instrumentedMethod));
+          StackManipulation.Size size = manipulation.apply(methodVisitor, implementationContext);
+          return new Size(size.getMaximalSize(), instrumentedMethod.getStackSize());
+        }
+      };
+    }
+
+    protected abstract StackManipulation invokeTargetMethod(MethodDescription instrumentedMethod);
+  }
+
+  private static final class ProcessElementDelegation extends MethodDelegation {
+    private static final Map<DoFnSignature.ProcessElementMethod.Parameter, MethodDescription>
+        EXTRA_CONTEXT_FACTORY_METHODS;
+
+    static {
+      try {
+        Map<DoFnSignature.ProcessElementMethod.Parameter, MethodDescription> methods =
+            new EnumMap<>(DoFnSignature.ProcessElementMethod.Parameter.class);
+        methods.put(
+            DoFnSignature.ProcessElementMethod.Parameter.BOUNDED_WINDOW,
+            new MethodDescription.ForLoadedMethod(
+                DoFn.ExtraContextFactory.class.getMethod("window")));
+        methods.put(
+            DoFnSignature.ProcessElementMethod.Parameter.INPUT_PROVIDER,
+            new MethodDescription.ForLoadedMethod(
+                DoFn.ExtraContextFactory.class.getMethod("inputProvider")));
+        methods.put(
+            DoFnSignature.ProcessElementMethod.Parameter.OUTPUT_RECEIVER,
+            new MethodDescription.ForLoadedMethod(
+                DoFn.ExtraContextFactory.class.getMethod("outputReceiver")));
+        EXTRA_CONTEXT_FACTORY_METHODS = Collections.unmodifiableMap(methods);
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+    }
+
+    private final DoFnSignature.ProcessElementMethod signature;
+
+    private ProcessElementDelegation(DoFnSignature.ProcessElementMethod signature) {
+      this.signature = signature;
+    }
+
+    @Override
+    protected StackManipulation invokeTargetMethod(MethodDescription instrumentedMethod) {
+      MethodDescription targetMethod =
+          new MethodCall.MethodLocator.ForExplicitMethod(
+                  new MethodDescription.ForLoadedMethod(signature.getTargetMethod()))
+              .resolve(instrumentedMethod);
+
+      // Parameters of the wrapper OldDoFn method:
```

Comment:
These aren't really parameters of the wrapper `OldDoFn`. The `OldDoFn.ProcessContext` has already been discarded, etc. Maybe just drop the comment for:

```
Available:
  - fn to delegate to, on the top of the stack
  - ExtraContextFactory for accessing additional parameters from the runner
```